hypothesis-python/src/hypothesis/core.py

```diff
@@ -160,8 +160,62 @@ class Example:
     reason: Any = field(default=None)
 
 
+# TODO_DOCS link to not-yet-existent patch-dumping docs
+
+
 class example:
-    """A decorator which ensures a specific example is always tested."""
+    """
+    Add an explicit input to a Hypothesis test, which Hypothesis will always
+    try before generating random inputs. This combines the randomized nature of
+    Hypothesis generation with a traditional parametrized test.
+
+    For example:
+
+    .. code-block:: python
+
+        @example("Hello world")
+        @example("some string with special significance")
+        @given(st.text())
+        def test_strings(s):
+            pass
+
+    will first call ``test_strings("Hello World")``, then
+    ``test_strings("some string with special significance")``, and then generate
+    inputs randomly.
```

Comment:
I think it might go innermost-first, actually?  I'd just say "... and ..., before generating inputs randomly".

I think this would be a good place to note that the decorators can be applied in any order, too.